allennlp/training/callbacks/log_writer.py

```diff
@@ -241,6 +241,13 @@ def log_batch(
                     self._batch_loss_moving_items[key]
                 )
 
+            for key, value in metrics.items():
+                if key.startswith("batch_"):
+                    continue
```

Comment:
Does this mean we don't log any metrics in `metrics` that start with `batch_`? Is that really what we want?

Reply:
I wanted to avoid creating `"batch_batch_loss"` and `"batch_batch_reg_loss"`, and I generalized a little. It doesn't make a ton of sense to define a metric called `batch_*`, because all metrics that come from the model are also calculated per epoch.

Reply:
`batch_size` is one exception I can think of

Reply:
Since we already handle `batch_loss` and `batch_reg_loss` separately, how we just explicitly ignore those here?

Reply:
Yeah, I'll do that as soon ad I can get GitHub to talk to me again :-/